static/app/views/performance/transactionSummary/transactionSpans/spanSummary/spanSummaryTable.tsx

```diff
@@ -0,0 +1,280 @@
+import {
+  type IndexedResponse,
+  SpanIndexedField,
+  type SpanMetricsQueryFilters,
+} from 'sentry/views/starfish/types';
+import {QueryParameterNames} from 'sentry/views/starfish/views/queryParameters';
+
+type DataRowKeys =
+  | SpanIndexedField.ID
+  | SpanIndexedField.TIMESTAMP
+  | SpanIndexedField.SPAN_DURATION
+  | SpanIndexedField.TRANSACTION_ID
+  | SpanIndexedField.TRACE
+  | SpanIndexedField.PROJECT;
+
+type ColumnKeys =
+  | SpanIndexedField.ID
+  | SpanIndexedField.TIMESTAMP
+  | SpanIndexedField.SPAN_DURATION;
+
+type DataRow = Pick<IndexedResponse, DataRowKeys> & {'transaction.duration': number};
+
+type Column = GridColumnHeader<ColumnKeys>;
+
+const COLUMN_ORDER: Column[] = [
+  {
+    key: SpanIndexedField.ID,
+    name: t('Span ID'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: SpanIndexedField.TIMESTAMP,
+    name: t('Timestamp'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: SpanIndexedField.SPAN_DURATION,
+    name: t('Span Duration'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+];
+
+const COLUMN_TYPE: Omit<
+  Record<ColumnKeys, ColumnType>,
+  'spans' | 'transactionDuration'
+> = {
+  span_id: 'string',
+  timestamp: 'date',
+  'span.duration': 'duration',
+};
+
+const LIMIT = 8;
+
+type Props = {
+  project: Project | undefined;
+};
+
+export default function SpanSummaryTable(props: Props) {
+  const {project} = props;
+  const organization = useOrganization();
+  const {spanSlug} = useParams();
+  const [spanOp, groupId] = spanSlug.split(':');
+
+  const location = useLocation();
+  const {transaction} = location.query;
+  const spansCursor = decodeScalar(location.query?.[QueryParameterNames.SPANS_CURSOR]);
+
+  const filters: SpanMetricsQueryFilters = {
+    'span.group': groupId,
+    'span.op': spanOp,
+    transaction: transaction as string,
+  };
+
+  const sort = useSpanSummarySort();
+
+  const {
+    data: rowData,
+    pageLinks,
+    isLoading: isRowDataLoading,
+  } = useIndexedSpans({
+    fields: [
+      SpanIndexedField.ID,
+      SpanIndexedField.TRANSACTION_ID,
+      SpanIndexedField.TIMESTAMP,
+      SpanIndexedField.SPAN_DURATION,
+      SpanIndexedField.TRACE,
+    ],
+    search: MutableSearch.fromQueryObject(filters),
+    limit: LIMIT,
+    referrer: 'api.performance.span-summary-table',
+    sorts: [sort],
+    cursor: spansCursor,
+  });
+
+  const transactionIds = rowData?.map(row => row[SpanIndexedField.TRANSACTION_ID]);
+
+  const eventView = EventView.fromNewQueryWithLocation(
+    {
+      name: 'Transaction Durations',
+      query: MutableSearch.fromQueryObject({
+        project: project?.slug,
+        id: `[${transactionIds?.join() ?? ''}]`,
+      }).formatString(),
+      fields: ['id', 'transaction.duration'],
+      version: 2,
+    },
+    location
+  );
+
+  const {
+    isLoading: isTxnDurationDataLoading,
+    data: txnDurationData,
+    isError: isTxnDurationError,
+  } = useGenericDiscoverQuery<
```

Comment:
can we run this only when we have results from the previous query so we're not making this request twice and having to cancel it in-flight?

Reply:
yeah good point, I initially tried to call the hook only after the previous query had data but React warns against using hooks conditionally. I looked into it just now and `useGenericDiscoverQuery` has an `enabled` option, so I'll update it to be enabled only when data from the previous query has loaded